game.py

```diff
@@ -0,0 +1,105 @@
+from collections import defaultdict
+
+from etc.utils import Color, read_words, get_random_word, is_valid_word
+from etc.consts import MAX_ATTEMPTS, INDEX_NOT_FOUND
+
+
+class WordleGame:
+    _WORDS = read_words()
+
+    def __init__(self):
+        self._goal_word = ""
+        self.attempts = 0
+        self.word_found = False
+        self.reset()
+
+    def apply_answer(self, guess):
+        if not is_valid_word(guess):
+            raise ValueError("Invalid word", guess)
+        elif self.attempts == MAX_ATTEMPTS:  # Lose
+            return  # Nothing
+        elif guess == self._goal_word:  # Win
+            self.word_found = True
+
+        self.attempts += 1
+
+        result = []
+        chars_to_idxs = self.__gen_chars_indexes_map()
+
+        for i, _ in enumerate(guess):
+            cur_char_color = self.__calc_char_color(guess, i, chars_to_idxs)
+
+            if cur_char_color is Color.GREEN:
+                chars_to_idxs[guess[i]].remove(i)
+            elif cur_char_color is Color.YELLOW:
+                alt_idx = self.__get_alternative_index(guess, chars_to_idxs[guess[i]])
+                chars_to_idxs[guess[i]].remove(alt_idx)
+            result.append(cur_char_color)
+
+        return result
+
+    def is_running(self):
+        return not self.word_found and self.attempts < MAX_ATTEMPTS
+
+    def reset(self):
+        self._goal_word = get_random_word(self._WORDS)
+        self.attempts = 0
+        self.word_found = False
+
+    def get_all_possible_words(self):
+        return self._WORDS.copy()
+
+    def is_winner(self):
+        return self.word_found
+
+    def get_answer(self):
+        return self._goal_word
+
+    def __gen_chars_indexes_map(self):
+        result = defaultdict(set)
+
+        for i, c in enumerate(self._goal_word):
+            result[c].add(i)
+
+        return result
+
+    def __calc_char_color(self, guess, char_idx, char_to_indexes_map):
+        """
+        Function receives a guess that was applied as an answer, and idx of a
+        letter that it's color is being evaluated and a map of chars to the
+        indexes they appear in the goal word.
+
+        There are 3 return values options:
+        1. GREEN: guess matches goal at the specified index
```

Comment:
Consider - you can use indactive result and map it to dict, so you wont need to explain so much in the code.
for example:
```python
results = {PERFECT:GREEN,
WRONG_PLACE:YELLOW,
NOT_PART_OF_THE_WORD:GREY,
}
```

Reply:
PERFECT, WRONG_PLACE, and NOT_PART_OF_THE_WORD should be consts? enums? something else?